### DISCUSSION TITLE
Waveshare  CAN/CAN FD Bus Data Analyzer?

### DISCUSSION THREAD

**Discussion by YuriExM** [collin80/SavvyCAN#752] [Q&A]
Hi guys, 
Can I ask if the Waveshare Industrial Grade CAN/CAN FD Bus Data Analyzer will work with SavvyCan ? 

https://www.waveshare.com/usb-can-fd.htm

Looking for USB adapter.

thank you !

**Comment by GPSdualSport** [comment]
I bought one from Amazon. It is not user friendly with Linux. Been fighting with it for several days now.
So far I can see the device is connected to USB using the **$ lsusb** command in the terminal window.
The wheels fall off the bus when **$ ip link show** command does not show any can0 socket.
Waveshare tech support did respond to my help request. They informed me this IS NOT A SOCKETCAN DEVICE. They also set a link to their 'Linux development demo' has a little python program that will make the LEDs on the front of the device flash and generate a couple lines of text proving the device 'works with Linux'. You just gotta love China based tech support.
There is a Amazon review from one person that claims it will work with Python-CAN and a reversed engineered interface. link below.
https://github.com/hardbyte/python-can/blob/main/doc/interfaces/canalystii.rst
No joy so far after loading Python-CAN and canalystii and double checking all the dependecies are up to date.
I'm a newbie with Linux and it's entirely possible someting simple is getting overlooked.
So the simple answer is 'no, the device does not work with SavvyCAN'. And so for it does not work with CAN-utils either. But that's just a newbie flounding with new tech.

**Comment by roema** [comment]
Waveshare provides a closed-source driver for this. It's not recognized as a socketCAN device. It uses a proprietary protocol called ZCAN. I bought the adapter and was quite annoyed, so I went to the trouble of writing a kernel module. It's still in beta, but it works for both channels. Currently, no CAN FD is implemented, only legacy CAN. But it works very well with SavvyCAN.

**Comment by roema** [comment]
Take a look at https://github.com/roema/ZCAN_USB

Works very well

Regards, Manu
